### PR TITLE
[TM] Add spec for TimePickerAndroid

### PR DESCRIPTION
**Pull request by jeanregisser**
## Summary

Part of #24875

## Changelog

[General] [Added] - Add TurboModule spec for TimePickerAndroid

## Test Plan

Run `yarn flow-check-ios` and `yarn flow-check-android` at the root level

### CODE REVIEW

File: Libraries/Components/TimePickerAndroid/NativeTimePickerAndroid.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+import type {
+  TimePickerOptions,
+  TimePickerResult,
+} from './TimePickerAndroidTypes';
+
+export interface Spec extends TurboModule {
+  +open: (options: TimePickerOptions) => Promise<TimePickerResult>;
+}
+
+export default TurboModuleRegistry.getEnforcing<Spec>('TimePickerAndroid');
```
Comment (ericlewis) [inline Libraries/Components/TimePickerAndroid/NativeTimePickerAndroid.js]: change this to only export if on android

File: Libraries/Components/TimePickerAndroid/TimePickerAndroid.android.js [hunk 2]
```diff
@@ -59,7 +58,7 @@ class TimePickerAndroid {
    * being undefined. **Always** check whether the `action` before reading the values.
    */
   static async open(options: TimePickerOptions): Promise<TimePickerResult> {
-    return TimePickerModule.open(options);
+    return NativeTimePickerAndroid.open(options);
```
Comment (ericlewis) [inline Libraries/Components/TimePickerAndroid/TimePickerAndroid.android.js]: after the first change you will need to do this:
```
if (NativeTimePickerAndroid) {
  return NativeTimePickerAndroid.open(options);
}
```

Comment (jeanregisser) [reply Libraries/Components/TimePickerAndroid/TimePickerAndroid.android.js]: Thanks, I've consolidated both iOS and Android implementations in https://github.com/facebook/react-native/pull/24897/commits/b45ea6179e0f334dcb444bae082d818e4dfa0b88

File: Libraries/Components/TimePickerAndroid/NativeTimePickerAndroid.js [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+'use strict';
+
+import type {TurboModule} from 'RCTExport';
+import * as TurboModuleRegistry from 'TurboModuleRegistry';
+import Platform from '../../Utilities/Platform';
+
+import type {
+  TimePickerOptions,
+  TimePickerResult,
+} from './TimePickerAndroidTypes';
```
Comment (RSNara) [inline Libraries/Components/TimePickerAndroid/NativeTimePickerAndroid.js]: Unfortunately, all types used by the spec need to be declared in this file. Perhaps we should relocate these types to this file and have files that depend on TimePickerAndroidTypes instead depend on NativeTimePickerAndroid?

File: Libraries/Components/TimePickerAndroid/TimePickerAndroid.js [hunk 4]
```diff
@@ -59,7 +58,13 @@ class TimePickerAndroid {
    * being undefined. **Always** check whether the `action` before reading the values.
    */
   static async open(options: TimePickerOptions): Promise<TimePickerResult> {
-    return TimePickerModule.open(options);
+    if (NativeTimePickerAndroid) {
+      return NativeTimePickerAndroid.open(options);
+    } else {
+      return Promise.reject({
```
Comment (uqmessias) [inline Libraries/Components/TimePickerAndroid/TimePickerAndroid.js]: :tada:

File: Libraries/Components/TimePickerAndroid/NativeTimePickerAndroid.js (hunk 3 above)
Comment (jeanregisser) [reply Libraries/Components/TimePickerAndroid/NativeTimePickerAndroid.js]: Sounds good, I did that in https://github.com/facebook/react-native/pull/24897/commits/03b99364f6f752c121fd502c116caee787cb7d4a

**Pull request closed without merging**